documentation/source/python_entry_point.rst

```diff
@@ -0,0 +1,28 @@
+Python Entry Point
+==================
+
+By default, cocotb starts the :class:`RegressionManager` as the testing framework for cocotb. However, there are many reason why a user might want to supplant the whole or part of the existing Python framework with an alternative. To this end, we have defined :envvar:`PYTHON_ENTRY`, to allow the user to enter into an alternative testing framework.
+
+.. versionadded:: 2.0
+
+How It Works
+------------
+
+After GPI implementations (VPI, VHPI, FLI, etc.) are registered, the Python implementation to the GPI (henceforth, PyGPI) starts a Python interpreter, loads the ``cocotb`` module, gets references to necessary helper functions from that module, and then calls into the ``_initialise_testbench`` function in that module to start the regression manager. The :envvar:`PYTHON_ENTRY` variable allows the user to specify a module and function other than ``cocotb._initialise_testbench`` that contains the alternative testing framework.
+
+How To Use It
+-------------
+
+.. currentmodule:: cocotb
+
+PyGPI expects the following functions to be implemented in the entry module. The names are required.
+
+.. autofunction:: _sim_event
+
+.. autofunction:: _log_from_c
+
+.. autofunction:: _filter_from_c
+
+Finally, there is the entry function itself. ``_initialise_testbench`` is being used as an example, the entry function is not required to be named the same. A single entry module can even contain multiple entry functions.
```

Comment:
Maybe it's just me, but could someone read this last sentence as "A single-entry module can even contain multiple entry functions." and be confused for a second?
Would "One entry module can even contain multiple entry functions." circumvent that and be still correct?

Reply:
I agree it could be ambiguous. What about "**The** entry module can even contain multiple entry functions."

Reply:
Indeed, that sounds better.